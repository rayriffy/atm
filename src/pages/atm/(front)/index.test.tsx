import { render, act } from '@testing-library/react'
import { describe, it, expect, vi, beforeEach } from 'vitest'

import Page from '.'

const propsSpy = vi.fn()
const { handlePadSpy, PinInputSpy, PadSpy } = vi.hoisted(() => ({
  handlePadSpy: vi.fn(),
  PinInputSpy: vi.fn(() => <div>PinInput</div>),
  PadSpy: vi.fn((props: unknown) => {
    propsSpy(props)
    return <div>Pad</div>
  }),
}))

vi.mock('$pad', () => ({
  Pad: PadSpy,
}))
vi.mock('$auth/handlePad', () => ({
  handlePad: handlePadSpy,
}))
vi.mock('$auth/pinInput', () => ({
  PinInput: PinInputSpy,
}))

describe('<Page />', () => {
  beforeEach(() => {
    vi.clearAllMocks()
  })

  it('should render page with all components', () => {
    const { container } = render(<Page />)

    expect(PinInputSpy).toBeCalledTimes(1)
    expect(PadSpy).toBeCalledTimes(1)

    expect(container).toMatchSnapshot()
  })

  it('should pass onClick to Pad', () => {
    render(<Page />)

    expect(PadSpy).toBeCalledTimes(1)

    act(() => {
      propsSpy.mock.calls[0][0].onClick()
    })

    expect(handlePadSpy).toBeCalledTimes(1)
  })
})
