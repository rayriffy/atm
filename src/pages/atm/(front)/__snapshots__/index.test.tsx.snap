// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Page /> > should render page with all components 1`] = `
<div>
  <div
    class="chakra-stack css-kmvjml"
  >
    <h2
      class="chakra-heading css-0"
    >
      ATM
    </h2>
    <div>
      PinInput
    </div>
  </div>
  <div>
    Pad
  </div>
</div>
`;
